features/nanostack/mbed-mesh-api/source/ThreadInterface.cpp

```diff
@@ -231,3 +231,13 @@ mesh_error_t Nanostack::ThreadInterface::device_pskd_set(const char *pskd)
 {
     return (mesh_error_t)thread_tasklet_device_pskd_set(pskd);
 }
+
+#define THREAD 0x2345
```

Comment:
Should this macro comparison magic be commented somewhere, eg. here? On https://github.com/ARMmbed/mbed-os/pull/6108#issuecomment-366940605 there was a good explanation for it.